### PR TITLE
[C++] Adding Oat++

### PR DESCRIPTION
**Pull request by cyrusmsk**
Link: https://oatpp.io

Previous issues:
- https://github.com/the-benchmarker/web-frameworks/issues/2426
- https://github.com/the-benchmarker/web-frameworks/issues/659

This PR should implement all 3 endpoints required by the benchmark.
I've tested locally, but after that made some changes in building scripts (config.yaml) so it will be possible to use it in main code. But it is better to test if everything is working properly.

### CODE REVIEW

**Comment by cyrusmsk** [comment]
It worked locally with Cmake 26.. I'm not sure if it will be fine for Cmake 18..
Maybe author of the framework could help with that.

**Comment by waghanza** [comment]
Thanks @cyrusmsk.

Maybe oat++ is available on Conan. 

cc @lganzzzo

**Comment by cyrusmsk** [comment]
@waghanza probably it is https://conan.io/center/recipes/oatpp?version=1.3.0

**Comment by cyrusmsk** [comment]
C++ is just a hell to make normal builds.. I really sorry for people who have to use it..
libevent is not going to work on bookworm - maybe just remove it then? Or fix build..

Regarding oatpp - it was finally built successfully, but had issues in testing:
Failures:

  1) routes POST : /user returns successfully
     Failure/Error: expect(response).to be_a(Net::HTTPSuccess)
       expected #<Net::HTTPNotFound 404 Not Found readbody=true> to be a kind of Net::HTTPSuccess
     # ./.spec/route_spec.rb:40:in `block (3 levels) in <top (required)>'

  2) routes POST : /user returns an empty string
     Failure/Error: expect(response.body.to_s).to be_empty
       expected `"server=oatpp/1.3.0\ncode=404\ndescription=Not Found\nmessage=No mapping for HTTP-method: 'POST', URL: '/user'\n".empty?` to be truthy, got false
     # ./.spec/route_spec.rb:44:in `block (3 levels) in <top (required)>'

Finished in 0.01723 seconds (files took 0.12152 seconds to load)
6 examples, 2 failures

Failed examples:

rspec ./.spec/route_spec.rb:39 # routes POST : /user returns successfully
rspec ./.spec/route_spec.rb:43 # routes POST : /user returns an empty string

@waghanza can you help with that? What does it mean?

File: cpp/oatpp/src/controller/MyController.hpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  
+   MyController(OATPP_COMPONENT(std::shared_ptr<ObjectMapper>, objectMapper) /* Inject object mapper */)
+    : oatpp::web::server::api::ApiController(objectMapper)
+  {}
+
+  ENDPOINT("GET", "/", root) {
+    return createResponse(Status::CODE_200, "");
+  }
+  
+  ENDPOINT("PUT", "/user", putUser) {
```
Comment (waghanza) [inline cpp/oatpp/src/controller/MyController.hpp:24]: Should be post

**Review by waghanza — CHANGES_REQUESTED** [review]

File: cpp/oatpp/src/controller/MyController.hpp (hunk 1 above)
Comment (cyrusmsk) [reply cpp/oatpp/src/controller/MyController.hpp]: Right. Thanks

File: cpp/oatpp/config.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+language:
+  version: 11
+framework:
+  github: oatpp/oatpp
+  version: 1.3.0
```
Comment (waghanza) [inline cpp/oatpp/config.yaml:5]:
```suggestion
  version: 1.3
```

File: cpp/oatpp/config.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  build_deps:
+
+  bin_deps:
+
+  binaries:
+    - build/oat_benchmark
+
+  download:
+    - git clone --depth=1 https://github.com/oatpp/oatpp.git
```
Comment (waghanza) [inline cpp/oatpp/config.yaml:15]:
```suggestion
    - git clone --depth=1 https://github.com/oatpp/oatpp.git -b 1.3.0
```

**Review by waghanza — APPROVED** [review]

**Pull request merged**